### PR TITLE
Small fixes to have nbdev work better with Google Colab notebooks

### PR DESCRIPTION
**Pull request by muellerzr**
Adjusts metadata to see if `tags` is present, if not pop
Adjusts small issue in `source` where it can be an array of a blank string
Adjusts another issue in source where a random `\r` would be placed (fixes https://github.com/fastai/nbdev/issues/355)
Added some tests

Also now the first properly cleaned nbdev straight from google colab!

Fixes https://github.com/fastai/nbdev/issues/363

cc: @jph00

### CODE REVIEW

**Comment by muellerzr** [comment]
Also going to try and handle this issue as well: https://github.com/fastai/nbdev/issues/355

TL;DR if you notice colab adds some /r randomly

**Comment by muellerzr** [comment]
Okay @jph00 this is ready for review, let me know your thoughts please

**Comment by muellerzr** [comment]
Converting to a draft for the time being. I think I can make this better. It doesn't account for if metadata is also in `outputs`, which apparently can happen.

File: nbdev/clean.py [hunk 1]
```diff
@@ -43,6 +43,12 @@ def clean_cell(cell, clear_all=False):
     if 'outputs' in cell:
         if clear_all: cell['outputs'] = []
         else:         clean_cell_output(cell)
+    if cell['metadata'].get('tags') is not None: cell['metadata'].pop('tags')
```
Comment (jph00) [inline nbdev/clean.py:46]: `pop('tags',None)` would let you remove the `if` IIUC

File: nbs/07_clean.ipynb [hunk 2]
```diff
@@ -168,6 +174,87 @@
     "               'source': 'awesome_code'})"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tst2 = {'cell_type': 'code',\n",
```
Comment (jph00) [inline nbs/07_clean.ipynb:183]: I'd suggesting simplifying this down to just the absolute minimum needed to test. Having lots of code in tests makes them hard to read and understand, and harder to maintain.

Comment (muellerzr) [reply nbs/07_clean.ipynb:183]: By moving this into `nbdev_clean_nbs` directly this test is no longer needed

**Comment by muellerzr** [comment]
@jph00 summary of changes:

`clean_cell_output` checks for `tags` in the metadata. Since we don't actually want tags anyways I left it's default to `o`

Since we also just pop off the `tags` anyways I just got rid of it in `cell_metadata_keep`

File: nbs/07_clean.ipynb [hunk 3]
```diff
@@ -168,6 +170,42 @@
...
+    "       'execution_count': 26,\n",
+    "       'metadata': {'tags':[]},\n",
+    "       'outputs': [{'execution_count': 2, \n",
+    "                    'data': {\n",
+    "                        'application/vnd.google.colaboratory.intrinsic+json': {\n",
```
Comment (jph00) [inline nbs/07_clean.ipynb:184]: There's still a lot of data here in the test. Can you minimize it to what's needed to test the logic?

**Review by jph00 — CHANGES_REQUESTED** [review]

File: nbs/07_clean.ipynb (hunk 3 above)
Comment (muellerzr) [reply nbs/07_clean.ipynb]: Let' me know if that's a better test

**Comment by jph00** [comment]
Thanks!

**Pull request merged**